lib/hasPermissions.ts

```diff
@@ -203,16 +207,63 @@ export class HasPermissionsDirectiveVisitor<
     const hasPermissionsValidateFunction: ValidateFunction<TContext> = (
       value: unknown,
       _: unknown,
-      __: unknown,
+      container: GraphQLArgument | GraphQLInputObjectType | GraphQLObjectType,
       context: TContext,
       resolverInfo: Record<string, unknown>,
       resolverSource: unknown,
       resolverArgs: Record<string, unknown>,
+      path?: Array<string>,
     ): unknown => {
+      if (
+        (location === DirectiveLocation.INPUT_FIELD_DEFINITION ||
+          location === DirectiveLocation.INPUT_OBJECT) &&
+        container instanceof GraphQLInputObjectType &&
+        path &&
+        path.length > 0 &&
+        container.getFields()[path[path.length - 1]] &&
+        Array.isArray(value)
+      ) {
+        const lastField = container.getFields()[path[path.length - 1]];
+
+        if (Array.isArray(lastField.defaultValue)) {
+          let verifyIfArraysAreEquals = true;
+
+          for (let i = 0; i < value.length; i += 1) {
+            if (value[i] !== lastField.defaultValue[i]) {
+              verifyIfArraysAreEquals = false;
+              break;
+            }
+          }
+
+          if (verifyIfArraysAreEquals) {
+            return value;
+          }
+        }
+      }
+
+      if (
+        (location === DirectiveLocation.INPUT_FIELD_DEFINITION ||
```

Comment:
we should try to be repetitive.
We can merge this if and the above one like this.
```
const lastField = container.getFields()[path[path.length - 1]];

if (
  (location === DirectiveLocation.INPUT_FIELD_DEFINITION ||
    location === DirectiveLocation.INPUT_OBJECT) &&
  container instanceof GraphQLInputObjectType &&
  path &&
  path.length > 0 &&
  lastField
) {
  if (value === lastField.defaultValue) return value;
  if (Array.isArray(lastField.defaultValue) &&
      Array.isArray(value) &&
      value.length === lastField.defaultValue.length &&
      !value.every((item, idx) => item === lastField.defaultValue[idx])) {
      return value;
  }
}
```

Reply:
Okay, I just need before check if path exists and container is instanceof GraphQLInputObjectType, and I think maybe this exclamation is wrong in:

`!value.every((item, idx) => item === lastField.defaultValue[idx])) {`,

because if all data is the same, it returns true and should stop checking permissions.